src/components/Snackbar/SnackbarContext.tsx

```diff
@@ -131,15 +131,42 @@ export const SnackbarProvider: React.FC = ({ children }) => {
       return null;
     }
 
+    const getPositionProps = (
+      position:
+        | 'top-left'
+        | 'top-middle'
+        | 'top-right'
+        | 'bottom-left'
+        | 'bottom-middle'
+        | 'bottom-right'
+        | undefined
+    ) => {
+      switch (position) {
+        case 'top-left':
+          return { top: 0, left: 6 };
+        case 'top-middle':
+          return { top: 0, left: '50%', style: { transform: 'translate(-50%, 0)' } };
+        case 'top-right':
+          return { top: 0, right: 6 };
+        case 'bottom-left':
+          return { bottom: 3, left: 6 };
+        case 'bottom-middle':
+          return { bottom: 3, left: '50%', style: { transform: 'translate(-50%, 0)' } };
+        case 'bottom-right':
+          return { bottom: 3, right: 6 };
+        default:
+          // bottom-left
+          return { bottom: 3, left: 6 };
+      }
```

Comment:
`[change]` Since this function is pure, we should move it out of the component. This will make sure it stays pure and doesn't introduce any over-rendering / stuff like that.

---

`[suggestion]` I think we can use an object for this, we'd also get the benefit of using the keys for TS

For example, something along these lines

```ts
const positionStyles = {
  'top-left': { top: 0, left: 6 },
  // ...etc
  'bottom-left': { bottom: 3, left: 6 }
}

// default position as a default argument, either in the component props or
// the utility function
const getPositionProps = (position: keyof positionStyles = 'bottom-left') => {
  return positionStyles[position]
}

// in the component
{...getPositionProps(transitions[0]?.item?.position)}
```